provisioning/baremetal_pod.go

```diff
@@ -559,6 +565,8 @@ func createContainerMetal3Httpd(images *Images, config *metal3iov1alpha1.Provisi
 			Capabilities: &corev1.Capabilities{
 				Drop: []corev1.Capability{"ALL"},
 			},
+			RunAsUser:  ptr.To(ironicUserID),
+			RunAsGroup: ptr.To(ironicGroupID),
```

Comment:
This container is marked as privileged, we shouldn't/cannot use specific users with it.